### DISCUSSION TITLE
Advanced methods to draw image parts to canvas

### DISCUSSION THREAD

**Discussion by projectitis** [flame-engine/flame#2668] [General]
Hi all,

I am trying to efficiently draw many part-images to a canvas, scaled differently in X and Y (including -ve scale, to flip it)
I have partial solutions, but none are quite right:

- Loop each part, and use `canvas.drawImageRect`
    - This will stretch each part correctly
    - But only +ve scale is supported (cannot flip)
    - And is inefficient because need to loop each part
- Use  `canvas.drawAtlas` to draw all at once
    - More efficient
    - Support -ve scale
    - But requires a uniform scale and does not support stretching differently in X and Y

Any ideas?

**Comment by spydon** [comment]
Not a solution, just a comment; you can do flips for the first solution if you do `canvas.save()` and `canvas.scale(-1)` before the  `drawImageRect` call and then `canvas.restore()` afterwards.

Since `RSTransform` only supports uniform scale I don't think that you'll be able to do it with `drawAtlas`.

**Reply by projectitis** [reply]
Yup, that's my conclusion.
My current implementation uses drawImageRect.
I had a `doh!` moment and realized I could use canvas transforms to achieve -ve scale, but haven't added that yet.
